src/common.jl

```diff
@@ -0,0 +1,122 @@
+function solve{uType,tType,isinplace,algType<:SundialsAlgorithm,F}(
+    prob::AbstractODEProblem{uType,tType,isinplace,F},
+    alg::Type{algType},timeseries=[],ts=[],ks=[];
+    callback=()->nothing,abstol=1/10^6,reltol=1/10^3,
+    saveat=Float64[],adaptive=true,maxiter=Int(1e5),
+    timeseries_errors=true,save_timeseries=true,
+    userdata=nothing,kwargs...)
+
+    tspan = prob.tspan
+    t0 = tspan[1]
+    T = tspan[end]
+
+    save_ts = sort(unique([t0;saveat;T]))
+
+    if T < save_ts[end]
+        error("Final saving timepoint is past the solving timespan")
+    end
+    if t0 > save_ts[1]
+        error("First saving timepoint is before the solving timespan")
+    end
+
+    if typeof(prob.u0) <: Number
+        u0 = [prob.u0]
+    else
+        u0 = vec(deepcopy(prob.u0))
+    end
+
+    sizeu = size(prob.u0)
+
+    ### Fix the more general function to Sundials allowed style
+    if !isinplace && (typeof(prob.u0)<:Vector{Float64} || typeof(prob.u0)<:Number)
+        f! = (t,u,du) -> (du[:] = prob.f(t,u); 0)
+    elseif !isinplace && typeof(prob.u0)<:AbstractArray
+        f! = (t,u,du) -> (du[:] = vec(prob.f(t,reshape(u,sizeu))); 0)
+    elseif typeof(prob.u0)<:Vector{Float64}
+        f! = prob.f
+    else # Then it's an in-place function on an abstract array
+        f! = (t,u,du) -> (prob.f(t,reshape(u,sizeu),reshape(du,sizeu));
+                          u = vec(u); du=vec(du); 0)
+    end
+
+    if alg == CVODE_BDF
+        mem = CVodeCreate(CV_BDF, CV_NEWTON)
+    elseif alg ==  CVODE_Adams
+        mem = CVodeCreate(CV_ADAMS, CV_FUNCTIONAL)
+    end
+
+    if mem == C_NULL
```

Comment:
`mem` doesn't look initialized so if alg is not CVODE_BDF or CVODE_Adams, this could be problematic?